src/VisualStudio/Core/Def/ProjectSystem/MiscellaneousFilesWorkspace.cs

```diff
@@ -122,6 +122,9 @@ public void RegisterLanguage(Guid languageGuid, string languageName, string scri
 
     private IEnumerable<MetadataReference> CreateMetadataReferences()
     {
+        // VisualStudioMetadataReferenceManager construction requires the main thread
+        _threadingContext.ThrowIfNotOnUIThread();
```

Comment:
Do we have a bug tracking fixing this already? It'd be good to link to this to make sure we don't forget to clean it up later.

Reply:
Went ahead and created something:

https://github.com/dotnet/roslyn/issues/77791